src/ut/alarm_trap_sender_test.cpp

```diff
@@ -0,0 +1,354 @@
+ * under which the OpenSSL Project distributes the OpenSSL toolkit software,
+ * as those licenses appear in the file LICENSE-OPENSSL.
+ */
+
+#include <unistd.h>
+#include <string>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "alarm_scheduler.hpp"
+#include "test_utils.hpp"
+
+#include "fakenetsnmp.h"
+#include "fakelogger.h"
+#include "test_interposer.hpp"
+
+using ::testing::_;
+using ::testing::A;
+using ::testing::Return;
+using ::testing::ReturnNull;
+using ::testing::InSequence;
+using ::testing::MakeMatcher;
+using ::testing::Matcher;
+using ::testing::MatcherInterface;
+using ::testing::MatchResultListener;
+using ::testing::SaveArg;
+using ::testing::Invoke;
+
+class EnterpriseTrapVarsMatcher : public MatcherInterface<netsnmp_variable_list*>
+{
+public:
+  explicit EnterpriseTrapVarsMatcher(oid trap_type,
+                                     std::string MIB_version,
+                                     std::string name,
+                                     oid alarm_oid,
+                                     oid resource_id,
+                                     std::string severity,
+                                     std::string description,
+                                     std::string details,
+                                     std::string cause,
+                                     std::string effect,
+                                     std::string action) :
+      _trap_type(trap_type),
+      _MIB_version(MIB_version),
+      _name(name),
+      _alarm_oid(alarm_oid),
+      _resource_id(resource_id),
+      _severity(severity),
+      _description(description),
+      _details(details),
+      _cause(cause),
+      _effect(effect),
+      _action(action) {}
+
+  virtual bool MatchAndExplain(netsnmp_variable_list* vl,
+                               MatchResultListener* listener) const {
+    
+    oid trap_type = *(vl->val.objid);
+    vl = vl->next_variable;
+    std::string MIB_version(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable;
+    std::string name(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable;
+    oid alarm_oid = *(vl->val.objid);
+    vl = vl->next_variable;
+    oid resource_id = *(vl->val.objid);
+    vl = vl->next_variable;
+    std::string severity(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable;
+    std::string description(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable; 
+    std::string details(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable; 
+    std::string cause(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable; 
+    std::string effect(reinterpret_cast<char const*>(vl->val.string));
+    vl = vl->next_variable; 
+    std::string action(reinterpret_cast<char const*>(vl->val.string));
+
+    return (_description == description) && (_details == details) && (_cause == cause)
```

Comment:
Is the solution to include a better DescribeTo function?
Something like *os << "trap has description" << _description << "and details" << _details << "and cause" << _cause

etc.